TBDataPreparation/202108_SPS/scripts/DoPhysicsConverter.py

```diff
@@ -8,15 +9,16 @@
 mrgfls = list(set(mrgfls) - set(recfls))
 phspath = "/eos/user/i/ideadr/TB2021_H8/recoNtuple/"
 
-if mrgfls:
-	print str(len(mrgfls))+" new files found"
-
+def doConversion(fname):
+    p1 = subprocess.Popen(["root","-l","-b","-q",".x","'PhysicsConverter.C("+fname+")'"])
+    p1.wait()
+    p2 = subprocess.Popen(["mv","physics_sps2021_run"+fname+".root",phspath])
+    p2.wait()
 
-for fl in mrgfls:
-	cmnd1 = "root -l -b -q .x 'PhysicsConverter.C(\""+fl+"\")'"
-	os.system(cmnd1)
-	cmnd2 = "mv physics_sps2021_run"+fl+".root "+phspath
-	os.system(cmnd2)
+if mrgfls:
+    print(str(len(mrgfls))+" new files found")
+    with multiprocessing.Pool(4) as pool:
+        pool.map_async(doConversion,mrgfls)
 
 if not mrgfls:
-	print "No new files found."
+    print("No new files found.")
```

Comment:
Works fine on my local PC. Probably it needs to call python3 instead of python or to source the right cvmfs env.
Currently I am using /cvmfs/fcc.cern.ch/sw/latest/setup.sh

Reply:
As we used python2 on lxlpus without sourcing any env let's try to keep python2 on. We have written several for the test beam, especially @ilnegri7, using python2. No need to switch to python3 for this one.